bayesian/utility/v2/component_analyses.py

```diff
@@ -112,7 +118,14 @@ def ca_validate_input(input_json: Dict, ecosystem: str) -> Tuple[List[Dict], Lis
     return packages_list, normalised_input_pkgs
 
 
-def get_batch_ca_data(ecosystem: str, packages) -> dict:
+def _fetcher_in_batches(func: Callable, packages: List,
+                        batch_size: int = COMPONENT_ANALYSES_SETTINGS.batch_size) -> Callable:
+    """Create partial function with batch package association."""
+    for i in range(0, len(packages), batch_size):
+        yield functools.partial(func, packages[i:i + batch_size])
```

Comment:
This seems like an incorrect use of partial, you are actually not creating a partial function here but are completing the function definition.

Also is yield required here, the list is being generated completely so does it make sense to create a generator here?

Reply:
Here the partial function used to bind the args which are batched into callable. List of callable passed into `map` which executes these callable in the context of thread executor.

in this case use of yield is not for lazy eval, but just to avoid list creation in the function which generates batch.